lib/client-assets.php

```diff
@@ -928,12 +928,15 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	 */
 	$allowed_block_types = apply_filters( 'allowed_block_types', true );
 
+	$post_type_object = get_post_type_object( $post_to_edit['type'] );
+
 	$editor_settings = array(
 		'alignWide'           => $align_wide || ! empty( $gutenberg_theme_support[0]['wide-images'] ), // Backcompat. Use `align-wide` outside of `gutenberg` array.
 		'availableTemplates'  => wp_get_theme()->get_page_templates( get_post( $post_to_edit['id'] ) ),
 		'blockTypes'          => $allowed_block_types,
 		'disableCustomColors' => get_theme_support( 'disable-custom-colors' ),
 		'disablePostFormats'  => ! current_theme_supports( 'post-formats' ),
+		'disablePreview'      => ! is_post_type_viewable( $post_type_object ),
```

Comment:
Can't we retrieve this information using the REST API?

Reply:
Unfortunately, it is not available there :(

Reply:
Can we create an issue to promote adding this to the REST API?

We're getting a bit too comfortable with cheating where the REST API should be providing necessary functionality.

Reply:
We can include this data on this `/wp-json/wp/v2/types` endpoint.

Reply:
Yes including it in the types seems like a good idea.